Source/Scene/Cesium3DTile.js

```diff
@@ -0,0 +1,1018 @@
+/*global define*/
+define([
+        '../Core/BoundingSphere',
+        '../Core/Cartesian3',
+        '../Core/Color',
+        '../Core/defaultValue',
+        '../Core/defined',
+        '../Core/defineProperties',
+        '../Core/destroyObject',
+        '../Core/getMagic',
+        '../Core/Intersect',
+        '../Core/joinUrls',
+        '../Core/JulianDate',
+        '../Core/loadArrayBuffer',
+        '../Core/Matrix3',
+        '../Core/Matrix4',
+        '../Core/Rectangle',
+        '../Core/Request',
+        '../Core/RequestScheduler',
+        '../Core/RequestType',
+        '../ThirdParty/when',
+        './Cesium3DTileChildrenVisibility',
+        './Cesium3DTileContentFactory',
+        './Cesium3DTileContentState',
+        './Cesium3DTileOptimizationHint',
+        './Cesium3DTileRefine',
+        './Empty3DTileContent',
+        './SceneMode',
+        './TileBoundingRegion',
+        './TileBoundingSphere',
+        './TileOrientedBoundingBox'
+    ], function(
+        BoundingSphere,
+        Cartesian3,
+        Color,
+        defaultValue,
+        defined,
+        defineProperties,
+        destroyObject,
+        getMagic,
+        Intersect,
+        joinUrls,
+        JulianDate,
+        loadArrayBuffer,
+        Matrix3,
+        Matrix4,
+        Rectangle,
+        Request,
+        RequestScheduler,
+        RequestType,
+        when,
+        Cesium3DTileChildrenVisibility,
+        Cesium3DTileContentFactory,
+        Cesium3DTileContentState,
+        Cesium3DTileOptimizationHint,
+        Cesium3DTileRefine,
+        Empty3DTileContent,
+        SceneMode,
+        TileBoundingRegion,
+        TileBoundingSphere,
+        TileOrientedBoundingBox) {
+    'use strict';
+
+    /**
+     * A tile in a 3D Tiles tileset.  When a tile is first created, its content is not loaded;
+     * the content is loaded on-demand when needed based on the view.
+     * <p>
+     * Do not construct this directly, instead access tiles through {@link Cesium3DTileset#tileVisible}.
+     * </p>
+     *
+     * @alias Cesium3DTile
+     * @constructor
+     */
+    function Cesium3DTile(tileset, basePath, header, parent) {
+        this._tileset = tileset;
```

Comment:
This constructor is kind of hard to follow.  For large contractors, we normally we put all of the constructor logic at the top of the function and then all of the public properties and then finally the private ones.